ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -345,23 +344,17 @@ private void recoverFailedCompactions(boolean remoteOnly) throws MetaException {
         HiveConf.ConfVars.HIVE_COMPACTOR_WORKER_TIMEOUT, TimeUnit.MILLISECONDS));
   }
 
-  private boolean foundCurrentOrFailedCompactions(ShowCompactResponse compactions, CompactionInfo ci) throws MetaException {
-    if (compactions.getCompacts() == null) {
+  private boolean foundCurrentOrFailedCompactions(CompactionInfo ci) throws MetaException {
+
+    ShowCompactRequest request =  new ShowCompactRequest();
+    request.setDbname(ci.dbname);
+    request.setTablename(ci.tableName);
+    request.setPartitionname(ci.partName);
+    final ShowCompactResponse currentCompactions = txnHandler.showCompact(request);
```

Comment:
I'm not sure this will be faster. Now you are filtering by DB, table, and partition, but on the other hand executing showcompact multiple times which has both serious DB and network (thrift) overhead. Did you do any measurements to check if this improves performance or not?